### DISCUSSION TITLE
Welcome to BTHomeV2-ESP32-example Discussions!

### DISCUSSION THREAD

**Discussion by Chreece** [Chreece/BTHomeV2-ESP32-example#10] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by robertocrr77** [comment]
Hello there,
I have tired to compile the code using first a ESP32-C3-Zero and a ESP32-C6-Zero but I will always get compiling errors. for example:
ESP32-C3-Zero
....BTHomeBuzzer\BTHome\BTHome.cpp: In member function 'void BTHome::buildPaket()':
....BTHomeBuzzer\BTHome\BTHome.cpp:250:52: error: 'class NimBLEAddress' has no member named 'getNative'
  250 |     const uint8_t *addrs = BLEDevice::getAddress().getNative();
      |                                                    ^~~~~~~~~
....BTHomeBuzzer\BTHome\BTHome.cpp:297:29: error: no matching function for call to 'NimBLEAdvertisementData::addData(std::string&)'
  297 |   oAdvertisementData.addData(payloadData);
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
In file included from ....libraries\NimBLE-Arduino\src/NimBLEAdvertising.h:35,
                 from ....libraries\NimBLE-Arduino\src/NimBLEServer.h:29,
                 from ....libraries\NimBLE-Arduino\src/NimBLEDevice.h:255,
                 from ....BTHomeBuzzer\BTHome\BTHome.cpp:2:
....libraries\NimBLE-Arduino\src/NimBLEAdvertisementData.h:29:10: note: candidate: 'bool NimBLEAdvertisementData::addData(const uint8_t*, size_t)'
   29 |     bool addData(const uint8_t* data, size_t length);
      |          ^~~~~~~
....libraries\NimBLE-Arduino\src/NimBLEAdvertisementData.h:29:10: note:   candidate expects 2 arguments, 1 provided
....libraries\NimBLE-Arduino\src/NimBLEAdvertisementData.h:30:10: note: candidate: 'bool NimBLEAdvertisementData::addData(const std::vector<unsigned char>&)'
   30 |     bool addData(const std::vector<uint8_t>& data);
      |          ^~~~~~~
....libraries\NimBLE-Arduino\src/NimBLEAdvertisementData.h:30:46: note:   no known conversion for argument 1 from 'std::string' {aka 'std::__cxx11::basic_string<char>'} to 'const std::vector<unsigned char>&'
   30 |     bool addData(const std::vector<uint8_t>& data);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~
....BTHomeBuzzer\BTHome\BTHome.cpp:314:17: error: 'class NimBLEAdvertising' has no member named 'setAdvertisementType'; did you mean 'setAdvertisementData'?
  314 |   pAdvertising->setAdvertisementType(BLE_GAP_CONN_MODE_NON);
      |                 ^~~~~~~~~~~~~~~~~~~~
      |                 setAdvertisementData

exit status 1

ESP32-C6-Zero
In file included from ....\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.3-a0f798cf\esp32c6/include/bt/include/esp32c6/include/esp_bt.h:17,
                 from ....\Documents\Arduino\libraries\NimBLE-Arduino\src/NimBLEDevice.h:21,
                 from ....\Documents\Arduino\BTHomeBuzzer\BTHome\BTHome.ino:1:
..../AppData/Local/Arduino15/packages/esp32/tools/esp32-arduino-libs/idf-release_v5.3-a0f798cf/esp32c6/include/bt/controller/esp32c6/esp_bt_cfg.h:19:10: fatal error: syscfg/syscfg.h: No such file or directory
   19 | #include "syscfg/syscfg.h"
      |          ^~~~~~~~~~~~~~~~~
compilation terminated.
exit status 1

**Comment by Chreece** [comment]
Seems that there are some changes to the [NimBLE-Arduino](https://github.com/h2zero/NimBLE-Arduino/commit/edc66e8b6c992d4992602c091bcee26261aa77d3) Library which aren't compatible with this example.
You can use an older version of the library.
Unfortunately I'm just a user that shared his project to the world (which also improved from a contributor) and I'm not longer interested on maintaining this project.
